src/request.cpp

```diff
@@ -145,25 +146,24 @@ set_impl(
     core::string_view t,
     http_proto::version v)
 {
-    detail::copied_strings cs(
-        this->buffer());
-    ms = cs.maybe_copy(ms);
-    t = cs.maybe_copy(t);
-
     auto const vs =
         to_string(v);
     auto const n =
+        // method SP
         ms.size() + 1 +
+        // request-target SP
         t.size() + 1 +
+        // HTTP-version CRLF
         vs.size() + 2;
-    auto dest = set_prefix_impl(n);
-    std::memcpy(
+    auto prefix_holder = set_prefix_impl(n);
+    auto dest = prefix_holder.prefix_.data();
+    std::memmove(
         dest,
         ms.data(),
         ms.size());
     dest += ms.size();
     *dest++ = ' ';
-    std::memcpy(
+    std::memmove(
```

Comment:
are you sure memmove is needed instead of memcpy?

Reply:
Yup. This is because this method is invoked with self-aliasing string_views so in this case, it's correct to use memmove.

If you switch it to memcpy, the sanitizer builds will fail with an invalid memcpy violation.

Reply:
lovely